articles/graph-data-structure-python-part1/index.md

```diff
@@ -0,0 +1,238 @@
+
+![Weighted Graphs](/engineering-education/graph-data-structure-python-part1/weighted-graph.png)
+
+*Figure: [Weighted Graph](https://www.clipartmax.com/middle/m2i8Z5i8G6b1b1b1_directed-weighted-ex1-directed-weighted-graph/)*
+
+##### Unweighted Graphs
+An unweighted graph does not have a value associated with every edge. An edge of an unweighted graph is represented as, `(u, v)`.
+
+- `u` -> Source vertex
+- `v` -> Destination vertex
+
+![Unweighted Graphs](/engineering-education/graph-data-structure-python-part1/unweighted-graph.png)
+
+*Figure:[Unweighted Graph](https://www.researchgate.net/figure/Unweighted-Undirected-Graph_fig3_327864836)*
+
+#### Special Graphs
+##### Trees
+An undirected graph with zero cycles is called a tree. A cycle in a graph is a sequence with the first and last vertices in the repeating sequence.
+
+It has X vertices and X-1 edges.
+
+![Tree - Graphs](/engineering-education/graph-data-structure-python-part1/tree-graph.png)
+
+*Figure:[Tree](https://en.wikipedia.org/wiki/Tree_(graph_theory))*
+
+##### Rooted Tree
+A rooted tree is a tree that has a designated root node. If edges point away from the root, it is called an **arborescence/out-tree**. If edges point towards the root, it is called an **anti-arborescence/in-tree**.
+
+![Rooted Tree](/engineering-education/graph-data-structure-python-part1/rooted-trees.png)
+
+*Figure: Rooted Tree*
+
+##### Directed Acyclic Graphs
+Directed Acyclic Graphs or DAGs are graphs with no directed cycles. They represent structures with dependencies. It's also important to note that: All [arborescences](https://en.wikipedia.org/wiki/Arborescence_(graph_theory)) are DAGs, but not all DAGs are arborescences.
+
+![Directed Acyclic Graph](/engineering-education/graph-data-structure-python-part1/dag.png)
+
+*Figure:[Directed Acyclic Graph](https://en.wikipedia.org/wiki/Directed_acyclic_graph)*
+
+##### Complete Graphs
+Complete graphs have a unique edge between every pair of vertices. A complete graph `n` vertices have `(n*(n-1)) / 2` edges and are represented by Kn.
+
+![Complete Graph](/engineering-education/graph-data-structure-python-part1/complete-graph.png)
+
+*Figure:[Complete Graph](https://www.geeksforgeeks.org/mathematics-graph-theory-basics/)*
+
+### Representing Graphs
+There are multiple ways of using data structures to represent a graph. The three most common ways are:
+
+- [Adjacency Matrix](#adjacency-matrix)
+- [Adjacency List](#adjacency-list)
+- [Edge List](#edge-list)
+
+#### Adjacency Matrix
+An Adjacency Matrix is a very simple way to represent a graph. In a weighted graph, the element `**A[i][j]**` represents the cost of moving from vertex `i` to vertex `j`.
+
+In an unweighted graph, the element `**A[i][j]**` represents a Boolean value that determines if a path exists from vertex `i` to vertex `j`. If A[i][j] == 0, then no path from vertex i to vertex j exists. If A[i][j] == 1, there is a path from vertex i to vertex j.
+
+##### Example:
```

Comment:
this example is great as it give an idea of how this is used. 
A sentences stating when this would be useful - would add value to the readers.